python/pyarrow/_compute.pyx

```diff
@@ -569,6 +543,10 @@ cdef class FunctionOptions(_Weakrefable):
         return self.get_options().Equals(deref(other.get_options()))
 
 
+def raise_invalid_function_option(value, description, *, error="keyword"):
```

Comment:
This is a private function, prefix it with an underscore?

Reply:
Do you think we should include it as a private method of `FunctionOptions` or leave it as a free standing function?

Reply:
Can leave it as a free standing function.